### PR TITLE
Patch 1 - Add FabledSkyblock Integration

### PR DESCRIPTION
**Pull request by DoctaEnkoda**

### CODE REVIEW

File: src/main/java/org/maxgamer/quickshop/integration/IntegrationHelper.java [hunk 1]
```diff
@@ -72,7 +73,9 @@ public void searchAndRegisterPlugins() {
         PluginManager pluginManager = plugin.getServer().getPluginManager();
         for (Map.Entry<String, Class<? extends IntegratedPlugin>> entry : integratedPluginNameMap.entrySet()) {
             String pluginName = entry.getKey();
-            if (pluginManager.isPluginEnabled(pluginName) && plugin.getConfig().getBoolean("integration." + pluginName.toLowerCase() + ".enable")) {
+            if (pluginName.equalsIgnoreCase("FabledSkyblock") && plugin.getConfig().getBoolean("integration." + pluginName.toLowerCase() + ".enable")) {
+                register(entry.getValue());
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/integration/IntegrationHelper.java:77]: A more ideal way is just add soft-depend in plugin.yml, so the plugin will enabled before quickshop enable.

File: src/main/java/org/maxgamer/quickshop/integration/FabledSkyblock/FabledIntegration.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+@IntegrationStage(loadStage = IntegrateStage.onEnableAfter)
+public class FabledIntegration extends QSIntegratedPlugin implements Listener {
+
+    private final boolean ignoreDisabledWorlds;
+    private final boolean whitelist;
+
+    public FabledIntegration(QuickShop plugin) {
+        super(plugin);
+        ignoreDisabledWorlds = plugin.getConfig().getBoolean("integration.fabledskyblock.ignore-disabled-worlds");
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/integration/FabledSkyblock/FabledIntegration.java:23]: Please add config update code in org.maxgamer.quickshop.QuickShop#updateConfig, also update `config-version` and config node in `config.yml`.

File: src/main/java/org/maxgamer/quickshop/integration/FabledSkyblock/FabledIntegration.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import org.bukkit.entity.Player;
+import org.bukkit.event.Listener;
+import org.jetbrains.annotations.NotNull;
+import org.maxgamer.quickshop.QuickShop;
+import org.maxgamer.quickshop.integration.IntegrateStage;
+import org.maxgamer.quickshop.integration.IntegrationStage;
+import org.maxgamer.quickshop.integration.QSIntegratedPlugin;
+
+@IntegrationStage(loadStage = IntegrateStage.onEnableAfter)
+public class FabledIntegration extends QSIntegratedPlugin implements Listener {
```
Comment (sandtechnology) [inline src/main/java/org/maxgamer/quickshop/integration/FabledSkyblock/FabledIntegration.java:16]: There have no listener code there, so it shouldn't implements Listener.

**Review by sandtechnology — CHANGES_REQUESTED** [review]

File: src/main/java/org/maxgamer/quickshop/integration/IntegrationHelper.java (hunk 1 above)
Comment (DoctaEnkoda) [reply src/main/java/org/maxgamer/quickshop/integration/IntegrationHelper.java]: its fine

**Comment by DoctaEnkoda** [comment]
fixed

File: src/main/resources/config.yml [hunk 4]
```diff
@@ -576,6 +576,12 @@ integration:
     #Should we delete the shop in the land when the shop owner lose the permission
     #in this land?
     delete-on-lose-permission: false
+  fabledskyblock:
```
Comment (sandtechnology) [inline src/main/resources/config.yml:579]: Please also add config update code in org.maxgamer.quickshop.QuickShop#updateConfig

**Review by sandtechnology — CHANGES_REQUESTED** [review]

**Comment by DoctaEnkoda** [comment]
it's good

**Comment by sandtechnology** [comment]
Thanks for contributing!

**Pull request merged**